### PR TITLE
Added emit method and modified jsdoc

### PR DESCRIPTION
**Pull request by EvidentlyCube**
I have added an 'emit' method because I thought it to be the easiest way to use `Hey` to have a global event communication in my app.

I have also improved the JSdoc to make it possible to use code hinting (at least in IntelliJ) and used more standard terminology in the descriptions.

### CODE REVIEW

**Comment by callmehiphop** [comment]
So I appreciate the thought, I never added an `emit()` method just because it's probably just as easy to .emit events using `$scope` or `$rootScope`.

I don't mind adding this in, but I'm going to ask that you do the following things:
1. Please format the white space to be what it was previously, it looks like everything got changed from two to four spaces and I'm having a hard time seeing what you actually changed since git thinks you just blew away the entire file.
2. Add unit tests! I know it seems trivial since it's just wrapping `$scope.$emit`, however the method itself is kinda trivial. So I'd like to see tests for it.
3. Updated the version from 0.0.5 to 0.0.3? What happened to 0.0.4?
4. Squash your commits!

Thanks!

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .DS_Store
 .tmp
 node_modules
-bower_components
\ No newline at end of file
+bower_components
+.idea
+HeyLibrary.iml
```
Comment (stephenplusplus) [inline .gitignore]: These types of files should be added in your global `gitignore`: http://stephenplusplus.github.io/me-but-you/#11-21-2013-globalize-your-gitignore

File: bower.json [hunk 2]
```diff
@@ -1,38 +1,39 @@
...
-    "url": "git://github.com/callmehiphop/hey.git"
-  },
-  "dependencies": {
-    "angular": "^1.0.7"
-  },
-  "devDependencies": {
-    "angular-mocks": "~1.2.15"
-  }
+    "name": "hey",
+    "version": "0.0.5",
```
Comment (stephenplusplus) [inline bower.json]: Never increase the version number in a PR. The maintainer will decide when to do that, and usually as a separate commit :)

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hey",
-  "version": "0.0.3",
+  "version": "0.0.6",
```
Comment (stephenplusplus) [inline package.json]: See above comment about version.

**Comment by EvidentlyCube** [comment]
Sure thing, I'll try to get these changes done and update the PR. The reason why increased the version number was because bower refused to use the github repo and I was at the moment on a slow internet connection and was not able to investigate it properly :)

**Comment by EvidentlyCube** [comment]
I am afraid I was unable to get the squash to work as intended (or I do not understand how it is supposed to work) so for the sake of simplicity I have just removed the commits and did a proper one.

Hopefully it will be more to your liking now and if not I'll be happy to do changes :). If you like these changes I'll also add the 'broadcast' method for full compatibility.

And the reason I wanted the 'emit' method was because it makes more semantical sense to use one class to both listen do and dispatch events, as opposed to listening with 'hey' but emitting with $rootScope or $scope.

File: bower.json [hunk 4]
```diff
@@ -23,7 +23,8 @@
     "pubsub"
   ],
   "authors": [
-    "Dave Gramlich <callmehiphop@gmail.com>"
+    "Dave Gramlich <callmehiphop@gmail.com>",
+    "Maurycy Zarzycki <maurycy.zarzycki@mauft.com> (http://retrocade.net)"
```
Comment (EvidentlyCube) [inline bower.json:27]: If you mind me adding myself here I can remove that line, I don't know what rules govern the JS community yet.

**Pull request merged**